pytorch_lightning/utilities/cloud_io.py

```diff
@@ -5,8 +5,11 @@
 
 
 def load(path_or_url: str, map_location=None):
-    parsed = urlparse(path_or_url)
-    if parsed.scheme == '' or Path(path_or_url).is_file():
-        # no scheme or local file
+    try:
+        is_path = urlparse(path_or_url).scheme == '' or Path(path_or_url).is_file()
+    except OSError:  # `is_file()` cannot parse URL on Windows and will raise error
+        is_path = False  # on OSError, `path_or_url` is supposed to be a URL
+    if is_path:
```

Comment:
Do we really need to catch such a severe OSError? Could be other things. 
If it really only concerns windows, I think it would be better coding practice to just check the platform with `platform.system() == "Windows"`
See also #2266

Reply:
Thank you. I then realized that using `Path.drive` doesn't require extra exception check. Would you mind reviewing the new commits?